app/controllers/nonsipp/BasicDetailsCheckYourAnswersController.scala

```diff
@@ -110,32 +114,68 @@ class BasicDetailsCheckYourAnswersController @Inject()(
     }
 
   def onSubmit(srn: Srn, mode: Mode): Action[AnyContent] = identifyAndRequireData(srn).async { implicit request =>
+    for {
```

Comment:
Yes you're right, this was a result of promptly pushing imperfectly - I put that part of the code up above to keep it separate while I worked on the rest of the code, but forgot to put it back underneath afterwards. I've made your recommended change and will push it as part of the next commit.

Reply:
Wouldn't this code, as is, send the audit event even if the submission has failed?

Reply:
Good spot, thank you! When moving the code around I must've incorrectly changed the nesting without realising - I'll fix this as part of the next series of commits.

Reply:
The latest commit ([a18b1b0](https://github.com/hmrc/pension-scheme-return-frontend/pull/763/commits/a18b1b0dee55b2b6c7c6cb069ad73783f20ae977)) has fixed this - I was able to test this by passing in a `None` and running the code locally, and the `.getOrRecoverJourney` code now correctly redirects the user to the Journey Recovery page, fixing the issue where it would ignore/discard this and continue to the Task List or Declaration page.

Reply:
Great!